### PR TITLE
Added remove set and get globalOptions interfaces

### PR DESCRIPTION
**Pull request by HANXU2018**
Added remove set and get globalOptions interfaces #180

### CODE REVIEW

File: jave-core/src/main/java/ws/schild/jave/Encoder.java [hunk 1]
```diff
@@ -466,6 +478,38 @@ public void encode(
...
+   * @throws EncoderException If a problems occurs during the encoding process.
+   */
+
+
+  public void encode(
+          List<MultimediaObject> multimediaObjects,
+          File target,
+          EncodingAttributes attributes,
+          EncoderProgressListener listener,
+          List<EncodingArgument> currOptions)
```
Comment (HANXU2018) [inline jave-core/src/main/java/ws/schild/jave/Encoder.java:511]: Sometimes transcoding requires some temporary parameters without changing the global configuration
So I added this parameter
` List<EncodingArgument> currOptions`

**Review by a-schild — APPROVED** [review]

**Comment by fengxiaochuang** [comment]
I am very lucky to see that the function I need has in the “pull-requests” area. 

Before the release of version 3.3.0,  package it or overwrite the class file by myself is the only fastest way, must not look like a best idea. 

Will you release a new version recently?

**Comment by paulvaneijden** [comment]
A new release is highly anticipated for our project

**Comment by a-schild** [comment]
Could you please test with the 3.3.0-SNAPSHOT release?
If it's working for you I will release 3.3.0

**Comment by paulvaneijden** [comment]
I have successfully tested the 3.3.0-SNAPSHOT release

Unfortunately it can't find the ffmpeg executable, even when I build it myself. The error is:
! Causing: java.io.IOException: Cannot run program "/var/folders/mq/_bwp9j397mn6cb9705jbb2hh0000gn/T/jave/ffmpeg-x86_64-3.3.0-SNAPSHOT-osx": error=2, No such file or directory

When I look in that folder I only see my (previously installed) 3.2.0 executable. I had to hardcode this version into DefaultFFMPEGLocator to override the ws.schild.jave.Version

When I do that I am able to successfully test the 3.3.0-SNAPSHOT release with the currOptions parameter with the current example code:

```
final EncodingArgument encodingImageArgument = new ValueArgument(ArgType.OUTFILE, "-frames:v", encodingAttributes -> Optional.of("1"));
encoder.encode(List.of(new MultimediaObject(uploadedFile)), targetImage, attrsImage, null, List.of(encodingImageArgument))
```

**Comment by a-schild** [comment]
What do you mean by "Unfortunately it can't find the ffmpeg executable, even when I build it myself. " and above you tested it sucessfully?

**Comment by paulvaneijden** [comment]
> What do you mean by "Unfortunately it can't find the ffmpeg executable, even when I build it myself. " and above you tested it sucessfully?

I mention that below that sentence :)

> When I look in that folder I only see my (previously installed) 3.2.0 executable. I had to hardcode this version into DefaultFFMPEGLocator to override the ws.schild.jave.Version
> 
> When I do that I am able to successfully test the 3.3.0-SNAPSHOT release

**Pull request merged**